### ISSUE TITLE
[ISSUE] Issue with `databricks_metastore` resource

### ISSUE THREAD

**Issue by patrickwilliamconway** [databricks/terraform-provider-databricks#2513]
### Configuration
```hcl
resource "databricks_metastore" "prod_metastore" {
  name          = "prod-databricks-metastore"
  storage_root  = "s3://prod-databricks-metastore/metastore"
}
```

### Expected Behavior

Create a metastore for my Databricks account. 

### Actual Behavior

```hcl
Error: cannot create metastore: invalid character '<' looking for beginning of value

  with databricks_metastore.prod_metastore,
  on root.tf line 131, in resource "databricks_metastore" "prod_metastore":
 131: resource "databricks_metastore" "prod_metastore" {
```

### Steps to Reproduce

```hcl
terraform apply
```

### Terraform and provider versions

Terraform and Provider Versions:
```zsh
> terraform version
Terraform v1.5.3
on darwin_arm64
+ provider registry.terraform.io/databricks/databricks v1.21.0
+ provider registry.terraform.io/hashicorp/aws v5.9.0
```
Provider Configurations:
```hcl
terraform {
  required_version = ">= 1.5.0"
  required_providers {
    databricks = {
      source  = "databricks/databricks"
      version = ">= 1.19.0, < 2.0.0"
    }
    aws = {
      source  = "hashicorp/aws"
      version = ">= 5.0.0, < 6.0.0"
    }
  }
}

provider "aws" {
  region = module.common.provider_region
}

provider "databricks" {
  host     = "https://accounts.cloud.databricks.com/"
  username = data.aws_ssm_parameter.username.value
  password = data.aws_ssm_parameter.password.value
  account_id = data.aws_ssm_parameter.account_id.value
}
```

### Important Factoids

I'm confused as to what the `databricks_metastore` is doing via the provider because the resource [here](https://github.com/databricks/terraform-provider-databricks/blob/v1.21.0/provider/provider.go#L109) points to [this](https://github.com/databricks/terraform-provider-databricks/blob/v1.21.0/catalog/resource_metastore.go#L71), and then `CREATE` it executes [this](https://github.com/databricks/terraform-provider-databricks/blob/v1.21.0/catalog/resource_metastore.go#L51), which is hitting an endpoint (`"/unity-catalog/metastores"`) different than the [public api documentation](https://docs.databricks.com/api/account/accountmetastores/create) (`
/api/2.0/accounts/{account_id}/metastores`).

I _am_ able to create a metastore using the REST api and the same credentials.

**Comment by patrickwilliamconway** [comment]
Okay, I figured this out after re-reading [this section](https://registry.terraform.io/providers/databricks/databricks/latest/docs/guides/unity-catalog#create-a-unity-catalog-metastore-and-link-it-to-workspaces) from the UC Guide Doc. I needed the `host` in provider to be a deployed workspace url. This doesn't make much sense to me why would this require a `workspace_url` to create an account level object? The REST API is operating at the account level (`/api/2.0/accounts/{account_id}/metastores`).

We have two separate ways of managing Databricks configurations - account level configs, and workspace level configs. AFAIU the whole point of UC is to have an account level, globally managed metastore. Requiring a deployed workspace url in the provider seems like an anti-pattern.

Is there a work around to use the Databricks provider to configure a metastore without needing a workspace URL? I want all of my infra managed via terraform, and it isn't great to have to have to switch to the REST API / Databricks UI to configure some parts.

**Comment by mgyucht** [comment]
We're working on it. The fact that metastore endpoints were exposed using a workspace-specific hostname was something that we did to enable API support for Unity Catalog while we were developing support at the account-level for REST APIs. We are gradually working to migrate workspace-level UC APIs to the account-level, and ultimately you should only use an account-level provider to manage UC resources. https://github.com/databricks/terraform-provider-databricks/pull/2182 is a WIP PR to fix this.

**Comment by patrickwilliamconway** [comment]
## Response
> We're working on it

Great! Thank you 🙏.

> something that we did to enable API support for Unity Catalog while we were developing support at the account-level for REST APIs

Makes sense - there are only so many people-hours in a day. 

> ultimately you should only use an account-level provider to manage UC resources

Got it, this makes sense. As for the error that I was getting:
```hcl
Error: cannot create metastore: invalid character '<' looking for beginning of value

  with databricks_metastore.prod_metastore,
  on root.tf line 131, in resource "databricks_metastore" "prod_metastore":
 131: resource "databricks_metastore" "prod_metastore" {
```
I think it would be helpful to have a more specific message, like "you're using an account url, you need to use a workspace url..." rather than what appears to be a generic golang error. Seems like that is coming from like a [json package](https://austindewey.com/2020/12/11/troubleshooting-invalid-character-looking-for-beginning-of-value/#:~:text=The%20first%20character%2C%20%3C%2C%20is,Hello%20world!)?

## Workaround
For anyone else that runs into this, I have a module dedicated to provisioning all resources for the metastore that pulls in a `reserve_workspace` to handle any API requirements like this. 

```hcl
provider "databricks" {
  alias = "reserve_workspace"
  host     = data.aws_ssm_parameter.reserve_workspace_url.value
  username = data.aws_ssm_parameter.account_admin_username.value
  password = data.aws_ssm_parameter.account_admin_password.value
  account_id = data.aws_ssm_parameter.account_id.value
}

resource "databricks_metastore" "prod_metastore" {
  provider = databricks.reserve_workspace
  name          = local.metastore_name
  storage_root  = "s3://${module.metastore_bucket.bucket_id}/${local.metastore_label}"
}
```

This is the same workaround as called out in the community post linked above.

**Comment by nkvuong** [comment]
fixed by #2182

**Comment by patrickwilliamconway** [comment]
thank you! I've upgraded provider versions and this is working. I appreciate it.
